api/src/test/java/org/openmrs/module/commonreports/reports/NewEpisodesOfDiseasesReportManagerTest.java

```diff
@@ -0,0 +1,182 @@
+		// verif
+		List<ReportDesign> designs = rs.getAllReportDesigns(false);
+		Assert.assertEquals(1, rs.getAllReportDesigns(false).size());
+		
+		ReportDesign design = designs.get(0);
+		Assert.assertEquals("7688966e-fca5-4fde-abab-1b46a87a1185", design.getUuid());
+		Assert.assertEquals(NewEpisodesOfDiseasesReportManager.REPEATING_SECTION,
+		    design.getProperties().get("repeatingSections"));
+		Assert.assertEquals(1, design.getResources().size());
+		
+		ReportDefinition def = design.getReportDefinition();
+		Assert.assertEquals("8b787bdc-c852-481c-b6fa-6683ec7e30d8", def.getUuid());
+	}
+	
+	@Test
+	public void testReport() throws Exception {
+		// setup
+		EvaluationContext context = new EvaluationContext();
+		context.addParameterValue("startDate", DateUtil.parseDate("2008-08-01", "yyyy-MM-dd"));
+		context.addParameterValue("endDate", DateUtil.parseDate("2009-09-30", "yyyy-MM-dd"));
+		boolean malariaVerified = false;
+		boolean feverVerified = false;
+		boolean diabetesVerified = false;
+		
+		// replay
+		ReportDefinition rd = manager.constructReportDefinition();
+		ReportData data = rds.evaluate(rd, context);
+		
+		// verify
+		for (Iterator<DataSetRow> itr = data.getDataSets().get(rd.getName()).iterator(); itr.hasNext();) {
+			DataSetRow row = itr.next();
+			if (row.getColumnValue("Maladies/Symptomes").equals("MALARIA")) {
+				assertEquals(ONE, row.getColumnValue("F_25-49"));
+				assertEquals(ONE, row.getColumnValue("M_1-4"));
+				assertEquals(ONE, row.getColumnValue("M_Total"));
+				assertEquals(ONE, row.getColumnValue("F_Total"));
+				malariaVerified = true;
+			}
+			if (row.getColumnValue("Maladies/Symptomes").equals("FEVER")) {
+				assertEquals(ONE, row.getColumnValue("M_1-4"));
+				assertEquals(ONE, row.getColumnValue("M_Total"));
+				assertNull(row.getColumnValue("F_Total"));
+				feverVerified = true;
+			}
+			if (row.getColumnValue("Maladies/Symptomes").equals("DIABETES")) {
+				assertEquals(ONE, row.getColumnValue("M_1-4"));
+				assertEquals(ONE, row.getColumnValue("M_Total"));
+				assertNull(row.getColumnValue("F_Total"));
+				diabetesVerified = true;
+			}
+		}
+		assertTrue(malariaVerified && feverVerified && diabetesVerified);
+	}
+	
+	private void updateDatabase(String filename) throws Exception {
+		Liquibase liquibase = getLiquibase(filename);
+		liquibase.update("Modify column datatype to longblob on reporting_report_design_resource table");
+		liquibase.getDatabase().getConnection().commit();
+	}
+	
+	private Liquibase getLiquibase(String filename) throws Exception {
+		Database liquibaseConnection = DatabaseFactory.getInstance()
+		        .findCorrectDatabaseImplementation(new JdbcConnection(getConnection()));
+		
+		liquibaseConnection.setDatabaseChangeLogTableName("LIQUIBASECHANGELOG");
+		liquibaseConnection.setDatabaseChangeLogLockTableName("LIQUIBASECHANGELOGLOCK");
+		
+		return new Liquibase(filename, new ClassLoaderResourceAccessor(getClass().getClassLoader()), liquibaseConnection);
+	}
```

Comment:
@Ruhanga , what are those for?

Reply:
In tests, the tables are built from their hibernate [`Entity`'s mappings](https://github.com/openmrs/openmrs-module-reporting/blob/7f3e0a44b2f417efa2f22a9b52ff6bf94813664d/api/src/main/resources/ReportDesign.hbm.xml#L48) whilst at runtime, [liquibase changesets](https://github.com/openmrs/openmrs-module-reporting/blob/7f3e0a44b2f417efa2f22a9b52ff6bf94813664d/api/src/main/resources/liquibase.xml#L180) are used. To test the design resource - the report template, I had to simulate an identical `reporting_report_design_resource` table to that at runtime.